DataServer/portAllocator.py

```diff
@@ -28,22 +50,39 @@ def reservePort(self) -> int:
         Returns:
             int: Next available port number
         """
-        
-        # TODO reimplement this
-        self.__current += 1
-        if self.__current > self.__end:
-            raise RuntimeError
-        return self.__current
+
+        for i in self.__portDict:
+            if self.__portDict[i]:
+                return_port = i
+                self.__portDict[i] == False
+            else:
+                continue
+
+        if return_port > self.__end:
+            raise RuntimeError('All ports occupied')
+        return return_port
```

Comment:
This logic does not check that the port is currently available.  Other services may dynamically obtain a socket from this block - it is not reserved exclusively for the ASM Data Server.